tests/chainer_tests/test_function.py

```diff
@@ -316,4 +316,50 @@ def test_forward_invalid(self):
             f(v)
 
 
+class TestFunctionDebug(unittest.TestCase):
```

Comment:
It is desirable if we can add test cases in which we check functions do not raise error when debug mode if `False`.